gwu-libraries/vopd
# 2024 revisions

Updates for 2024 election period:

- Adds a script and CLI for gathering data through BrightData, the company that Professor Bahador and his team are partnering with to collection. Also adds ENV file setup necessary for BrightData API token. 
- Add `BDTwitterDocumentSet` class in `document.py` for generating a document set from the results of aforementioned API service,  as the results differ slightly from SFM gathered data. 
- Add `bdtweets` as a mode for main `vopd.py` CLI and processes tweets from a CSV file of tweets gathered through BrightData. 
- Update readme to reflect changes. 

To test:
- Run `python vopd.py --subjectfile groups.csv --keywordfile sentiment_words.csv --mode bdtweets sample_data.csv`
- Ensure output file (`bd-extracts-tweets.csv`) is generated with columns for:
  - Date
  - Handle
  - Name
  - Party (empty)
  - House/Senate (empty)
  - Incumbent/Challenger (empty)
  - Tweet
  - Sentiment Word
  - Group Term
  - Sentiment Tone (N/P)
  - Group Type (C, I, F, O or P)
  - Coding Decision
  - Extract
